scalr/resource_scalr_run_schedule_rule.go

```diff
@@ -0,0 +1,136 @@
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"schedule": {
+				Description: "Cron expression for scheduled runs. Time should be in UTC.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"schedule_mode": {
+				Description: "Mode of the scheduled run. (\"apply\", \"destroy\", \"refresh\")",
+				Type:        schema.TypeString,
+				Required:    true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(scalr.ScheduleModeApply),
+					string(scalr.ScheduleModeDestroy),
+					string(scalr.ScheduleModeRefresh),
+				}, false),
+			},
+			"workspace_id": {
+				Description: "The identifier of the Scalr workspace, in the format `ws-<RANDOM STRING>`.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceScalrRunScheduleRuleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+	id := d.Id()
+
+	log.Printf("[DEBUG] Read run schedule rule: %s", id)
+	rule, err := scalrClient.RunScheduleRules.Read(ctx, id)
+	if err != nil {
+		if errors.Is(err, scalr.ErrResourceNotFound) {
+			log.Printf("[DEBUG] Run schedule rule %s not found", id)
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Error reading run schedule rule %s: %v", id, err)
+	}
+
+	// Update config.
+	_ = d.Set("schedule", rule.Schedule)
+	_ = d.Set("schedule_mode", rule.ScheduleMode)
+	_ = d.Set("workspace_id", rule.Workspace.ID)
+	return nil
+}
+
+func resourceScalrRunScheduleRuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+
+	// Create a new options struct.
+	options := scalr.RunScheduleRuleCreateOptions{
+		Schedule:     d.Get("schedule").(string),
+		ScheduleMode: scalr.ScheduleMode(d.Get("schedule_mode").(string)),
+	}
+
+	// Get and check the workspace.
+	if workspaceID, ok := d.GetOk("workspace_id"); ok {
+		ws, err := scalrClient.Workspaces.ReadByID(ctx, workspaceID.(string))
+		if err != nil {
+			return diag.Errorf(
+				"Error retrieving workspace %s: %v", workspaceID, err)
+		}
+		options.Workspace = ws
+	}
+
+	log.Printf("[DEBUG] Create run schedule rule %s %s for workspace %s", options.ScheduleMode, options.Schedule, options.Workspace.ID)
+	rule, err := scalrClient.RunScheduleRules.Create(ctx, options)
+	if err != nil {
+		return diag.Errorf(
+			"Error creating run schedule rule %s %s for workspace %s", options.ScheduleMode, options.Schedule, options.Workspace.ID)
+	}
+	d.SetId(rule.ID)
+
+	return resourceScalrRunScheduleRuleRead(ctx, d, meta)
+}
+
+func resourceScalrRunScheduleRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	scalrClient := meta.(*scalr.Client)
+
+	id := d.Id()
+	if d.HasChange("schedule") || d.HasChange("schedule_mode") {
+		opts := scalr.RunScheduleRuleUpdateOptions{
+			ID:           id,
```

Comment:
I would skip setting `ID` here to avoid confusion — keep only updatable attributes.